paywall/src/__tests__/reducers/transactionsReducer.test.js

```diff
@@ -55,27 +54,6 @@ describe('transaction reducer', () => {
     ).toBe(initialState)
   })
 
-  // Upon changing account, we need to clear the existing transaction. The web3 middleware will
-  // re-populate them
-  it('should clear the transactions when receiving SET_ACCOUNT', () => {
```

Comment:
we do check for this. The key lines are in `Paywall.js`:

```js

  const lockKey = Object.values(keys).find(
    key =>
      account &&
      lock &&
      key.lock === lock.address &&
      key.owner === account.address
  )

  // Let's select the transaction corresponding to this key purchase, if it exists
  // This transaction is of type KEY_PURCHASE
  transaction = Object.values(transactions).find(
    transaction =>
      transaction.type === TRANSACTION_TYPES.KEY_PURCHASE &&
      transaction.key === (lockKey && lockKey.id)
  )
```

This code first compares the existing key owner to the current account, and then uses that to retrieve transactions, so we ignore anything that does not match the current account

Reply:
Well that's the thing, I think it is fragile to do such a check in the UI because if we introduce a new feature that requires the use of transactions we will need to re-implement the same logic. I think it is much better to know that the data in the redux store is "trusted"?

Reply:
Agreed. The problem is we only really need to reset transaction information when the account address is changed, but SET_ACCOUNT is used for any change, and we don't have access in the transactions reducer to the account to determine whether this is true. The better solution may be to split off account change actions from account balance updates?

Reply:
Exactly